src/directive.ts

```diff
@@ -96,9 +98,15 @@ class AttributePart {
 
   constructor(legacyPart: legacyLit.AttributePart|legacyLit.PropertyPart) {
     this.legacyPart = legacyPart;
-    this.type = (legacyPart instanceof legacyLit.PropertyPart) ?
-        PartType.PROPERTY :
-        PartType.ATTRIBUTE;
+  }
+}
+
+export type{PropertyPart};
+class PropertyPart extends AttributePart {
+  readonly type = PartType.PROPERTY;
+
+  constructor(legacyPart: legacyLit.PropertyPart) {
```

Comment:
fwiw, you don't need to write this constructor, even though leaving it out won't change the emit since the class needs a field initializer.

Reply:
Thanks, removed